src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -813,7 +815,7 @@ void SpatialPooler::updatePermanencesForColumn_(vector<Real>& perm,
   numConnected = 0;
   for (UInt i = 0; i < perm.size(); ++i)
   {
-    if (perm[i] >= synPermConnected_)
+    if (perm[i] >= synPermConnected_ - PERMANENCE_EPSILON)
```

Comment:
As we discussed, you can think of floating point math as creating a bell curve of possible results. If the "correct" answer is 0.5, the floating point math might result in numbers between 0.4999996 and 0.5000004. With this EPSILON, we move the threshold so that it sits on one side of the entire bell curve, so results will be consistent no matter where it landed in the bell curve.

I do think `>=` makes sense, because then the code still works correctly if `PERMANENCE_EPSILON` is set to 0.